### ISSUE TITLE
Next display shortcut unresponsive (permanently greyed out in menu) 

### ISSUE THREAD

**Issue by corneliusroemer** [rxhanson/Rectangle#55]
macOS version: macOS 10.15.2 (19C57)
Rectangle version: Version 0.14 (18)

Expected Behaviour: I can move a window to another display with shortcut ctrl+option+cmd+-> or by clicking the menu item for it.

Observed Behaviour: The menu item is greyed out, when pressing the shortcut, the window is moved in some unexpected way (if it's full width/height, nothing happens, otherwise it moves to the centre without changing proportions).

To be noted: I had a similar issue with spectacle, which is why I switched. On spectacle, I heard a "pling" sound (error like) when trying to move window to next display. 

When I reprogram the shortcut, it's still greyed out - what does this tell me as a user? It may be there's an issue somewhere that I don't understand correctly.

I've tested with various apps: Spotify, Chrome, Finder, it doesn't seem to be specific. 

Here's a screenshot of the menu item greyed out.

<img width="250" alt="Screenshot 2019-12-27 at 20 18 54" src="https://user-images.githubusercontent.com/25161793/71529754-b3104b00-28e6-11ea-9338-6a7ddbadf384.png">

Here are the logs, it seems like my (soft) Display 2 is not recognised as being present. Is that meant to be that way?
```
2019-12-27T20:30:37+01:00: AX position proposed: (600.0, 373.0), result: (600.0, 373.0)
2019-12-27T20:30:37+01:00: AX sizing proposed: (480.0, 327.0), result: (480.0, 327.0)
2019-12-27T20:30:37+01:00: nextDisplay | display: (0.0, 0.0, 1680.0, 1027.0), calculatedRect: (600.0, 373.0, 480.0, 327.0), resultRect: (600.0, 373.0, 480.0, 327.0), srcScreen: Built-in Retina Display, destScreen: Built-in Retina Display, resultScreen: Built-in Retina Display
```

**Comment by rxhanson** [comment]
You are correct that your display is not recognized as being present (that's the scenario for greyed out next/prev display). This is direct from Apple's screen APIs, so there's possibility that soft displays can't be supported... what is a soft display?

My expected behavior from Rectangle is actually that it should act the same way as Spectacle, with a beep since it can't do it.

**Comment by corneliusroemer** [comment]
I'm sorry, I must have been unclear. I meant that I can't move a window to the next "desktop" that is part of Mac's window management - not a real desktop, the one you can see when you go to Window management, or through a hot corner. I might not know the right technical name. Here's a screenshot of what I mean.

Maybe the behaviour is as expected - in which case this is a feature requests/suggestion. It would be handy to be able to move the window over to desktop 2 by shortcut - corresponding to this Stackexchange thread: https://superuser.com/questions/184763/is-there-a-way-to-move-the-current-window-to-another-desktop-without-using-a-mou

<img width="476" alt="Screenshot 2019-12-29 at 09 57 12" src="https://user-images.githubusercontent.com/25161793/71554621-ec99a100-2a21-11ea-9b12-27dbbc1c3963.png">

**Comment by rxhanson** [comment]
Update: It's still not something I have a strong desire to add into Rectangle, but I would be fine merging in a pull request if anyone else wanted to implement it and put it behind a terminal command config (not in the UI).

This was added into [Rectangle Pro](https://rectangleapp.com/pro) (paid). 

-------------

Original Answer:
Ah, ok. Spaces (or Desktops) is a feature request that I've expected to come up at some point. The short answer is that I don't have plans to add this in, sorry.

Here's the long answer. Apple doesn't have a public API available for Spaces, there's only a private API. Amethyst and SizeUp both use the private API to provide this feature, and last time I tried it in SizeUp it didn't work for me. The choice not to use private APIs in Rectangle is the choice of more stability over more features. Since Amethyst is open source (and written in Swift), someone could work this feature in to a fork of Rectangle if they want without having to figure it out from scratch. I don't mind providing details on how to add more window actions for next/previous Space if anyone is interested, but again, this is not something I would officially roll in to Rectangle. It's also a decent amount of code to add in (and maintain).

If Apple decided to roll out a public API for interacting with Spaces, I would reconsider adding these in.
